docs/DeveloperGuide.md

```diff
@@ -1,29 +1,159 @@
 # Developer Guide
+The aim of this guide is to help readers understand how the system and components of RecipEditor 
+is designed, implemented and tested. This developer guide also serves to help 
+developers to understand the architecture of RecipEditor and some design considerations. 
+Click to view the latest release of [RecipEditor]((https://github.com/AY2223S1-CS2113-T18-2/tp/releases)).
+
+## Content page
+[Acknowledgements](#acknowledgements)
+
+[Design](#design)
+- [Architecture](#architecture)
+- [Logic Component](#logic-component)
+- [Ui Component](#ui-component)
+- [Storage Component](#storage-component)
+- [Implementation](#implementation)
+    - [Loading Of Data On Startup](#loading-of-data-on-startup)
+    - [Parsing of Commands](#parsing-of-commands)
+    - [Add Recipe](#add-recipe)
+    - [Edit an Existing Recipe](#add-an-existing-recipe)
+    - [Find Recipe Based on Recipe Name and Ingredient](#find-recipe-based-on-recipe-name-and-ingredient)
+- [Product Scope](#product-scope)
+    - [Target User Profile](#target-user-profile)
+    - [Value Proposition](#value-proposition)
+- [User Stories](#user-stories)
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+### External Libraries
+- org.apache.commons:commons-lang3:3.0 [link](https://mvnrepository.com/artifact/org.apache.commons/commons-lang3/3.0)
+- org.apiguardian:apiguardian-api:1.1.0 [link](https://mvnrepository.com/artifact/org.apiguardian/apiguardian-api)
 
-## Design & implementation
+## Design
+### Architecture
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<p align="center" width="100%">
+  <img width="80%" src="images/Architecture.png" alt="Storage Class Diagram"/>
+</p>
 
+`Recipeditor` calls to various class a perform all the tasks assigned by the user.
+
+- `Ui`: handles interactions with users, including printing and reading
+- `Storage`: manages the storage of the list of recipes
+- `Command`: instructs the current task to perform
+- `CommandResult`: explains the outcome of each command performed
+- `Parser`: interprets the user input into different commands
+
+Software running flow:
+
+Upon start, Recipeditor will check load or create saves.
+
+During software run, it will repeat iterations of reading and executing commands.
+In each iteration,raw user inputs are read from CLI. They will be
+interpreted into commands for the software to execute.
+Each execution will either write or read the list of recipes
+depending on the command. Finally, the result of current iteration
+is reflected to the user.
+
+Termination of software purges all temporary data, while saved changes
+can be loaded from saves upon next software launch.
+
+
+### Ui Component
```

Comment:
The class diagram styling is wrong. There is not supposed to be a C beside the object.